tenant-password-decryptor/src/main/java/org/wso2/custom/password/decrypt/AsymmetricKeyDecryptImpl.java

```diff
@@ -0,0 +1,172 @@
+package org.wso2.custom.password.decrypt;
+
+import javax.crypto.Cipher;
+import java.io.FileInputStream;
+import java.nio.charset.Charset;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.Security;
+import java.util.Scanner;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import org.apache.axiom.om.util.Base64;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+
+
+public class AsymmetricKeyDecryptImpl {
+
+    private static Gson gson = new Gson();
+
+    public static void main(String[] args) throws Exception {
+        Scanner scanner = new Scanner(System.in);
+        String plainText;
+        if (args.length == 4) {
+            plainText = decrypt(Base64.decode(args[0]), args[1], args[2], args[3]);
+        } else {
+            System.out.print("Encrypted Text : ");
+            String encryptedText = scanner.next();
+            System.out.print("KeyStore file path : ");
+            String keystorePath = scanner.next();
+            System.out.print("KeyStore alias : ");
+            String alias = scanner.next();
+            System.out.print("KeyStore password : ");
+            String password = scanner.next();
+            plainText = decrypt(Base64.decode(encryptedText), keystorePath, alias, password);
+        }
+
+        System.out.println("*** Plain Text ***\n" + plainText + "\n******************");
+    }
+
+    private static String decrypt(byte[] ciphertext, String keystore, String keystoreAlias, String keystorePassword) throws Exception {
+        KeyStore keyStore = getKeyStore(keystore, keystorePassword);
+        PrivateKey privateKey = (PrivateKey)keyStore.getKey(keystoreAlias, keystorePassword.toCharArray());
+        Security.addProvider(new BouncyCastleProvider());
+        Cipher keyStoreCipher;
+        String cipherTransformation = "RSA/ECB/OAEPwithSHA1andMGF1Padding";
+        CipherHolder cipherHolder = cipherTextToCipherHolder(ciphertext);
+        if (cipherHolder != null) {
+            // cipher with meta data.
+            System.out.println("Cipher transformation for decryption : " + cipherHolder.getTransformation());
+            keyStoreCipher = Cipher.getInstance(cipherHolder.getTransformation(), "BC");
+            ciphertext = cipherHolder.getCipherBase64Decoded();
+        } else {
+            keyStoreCipher = Cipher.getInstance(cipherTransformation, "BC");
+        }
+
+        keyStoreCipher.init(Cipher.DECRYPT_MODE, privateKey);
+        return new String(keyStoreCipher.doFinal(ciphertext), Charset.defaultCharset());
+    }
+
+    private static KeyStore getKeyStore(String keystore, String keystorePassword) throws Exception {
```

Comment:
add new line after the method definition

Reply:
Fixed with 1e3ad1cdc00d45759c43054533ac093caf76ea8d